### PR TITLE
[PXZ-20] Product Management

### PR DESCRIPTION
**Pull request by juliabaptista**
# [PXZ-20] Product Management

## Technical Tasks:
- [x] Product Model
- [x] Product Repository
- [x] Product Controller
- [x] Product Service
- [x] Database Configuration for Products
- [x] Implement GET endpoints - Product Listing
- [x] Pageable in Spring Data - Handle server-side pagination

## Technical Tasks - NEED TO BE IMPLEMENTED:
- [ ] Implement POST endpoint
- [ ] Implement PUT endpoint
- [ ] Implement DELETE endpoint
- [ ] Validation and error handling for product

## Postman outputs for product listing:

> **GET endpoint - get all products with pagination support** <br />

Screenshot 2024-02-12 at 19 55 03 <br />
<br />

> **GET endpoint - get products by id** <br />

Screenshot 2024-02-12 at 19 55 29 <br />
<br />

> **GET endpoint - get products by keyword** <br />

Screenshot 2024-02-12 at 19 55 54 <br />
<br />

> **Error handling:** <br />

Screenshot 2024-02-12 at 19 52 54
<br />
Screenshot 2024-02-12 at 19 53 24
<br />

[PXZ-20]: https://jbptst.atlassian.net/browse/PXZ-20?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/de/zalando/MockData.java
Comment (faruka-startsteps) [inline src/main/java/de/zalando/MockData.java:1]: Make sure to delete this later

File: src/main/java/de/zalando/controller/ProductController.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      @RequestParam("keyword") String keyword,
+      @RequestParam(value = "page", defaultValue = "0") int page,
+      @RequestParam(value = "size", defaultValue = "10") int size
+  ) {
+    try {
+      return ResponseEntity.ok(
+          productService.getAllByArchivedIsFalseAndProductNameContainingIgnoreCase(keyword, page,
+              size));
+    } catch (ProductNotFoundException e) {
+      return ResponseEntity.status(HttpStatus.NOT_FOUND).body(e.getMessage());
```
Comment (faruka-startsteps) [inline src/main/java/de/zalando/controller/ProductController.java:52]: should be handled in a global exception handler if you have one, throwing the exception from the service, and catching it with the handler

File: src/main/java/de/zalando/controller/ProductController.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    }
+  }
+
+  @GetMapping("/{id}")
+  public ResponseEntity<?> getProductByProductIdAndArchivedIsFalse(
+      @PathVariable("id") Long productId) {
+    try {
+      return ResponseEntity.ok(productService.getProductByProductIdAndArchivedIsFalse(productId));
+    } catch (ProductNotFoundException e) {
```
Comment (faruka-startsteps) [inline src/main/java/de/zalando/controller/ProductController.java:61]: same

File: src/main/java/de/zalando/model/entities/User.java [hunk 3]
```diff
@@ -25,4 +25,6 @@ public class User {
   @Enumerated
   @Column(name = "role")
   private Role role;
+  @Column(name = "is_archived", columnDefinition = "false")
```
Comment (faruka-startsteps) [inline src/main/java/de/zalando/model/entities/User.java:28]: I think the `false` value represents a default value for `is_archived` and this is not how it is done. You can just assign false to the archived field here and change it later where you need to change it

**Review by faruka-startsteps — APPROVED** [review]
Small comments

File: src/main/java/de/zalando/MockData.java
Comment (juliabaptista) [reply src/main/java/de/zalando/MockData.java:1]: Yes, definitely! :) @faruka-startsteps

**Pull request merged**